### PR TITLE
feat: add timeout input to agent and main workflows

### LINKED ISSUE #78: add timeout-minutes options for pipeline jobs
## The Problem

Sometimes the workflow hangs for large amounts of time (>1hour) when some of the actions fail.

We would like to be able to configure the timeout-minutes configuration for the pipeline jobs.

## Additional Resources

- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

### PR DESCRIPTION
**Pull request by johndalvik**
We had a problem when running the workflows, one of the agents failed but somehow the main job stuck, and was waiting for the agent to finish/report back but it was running for an hour before we noticed and aborted. (Default timeout for GitHub workflow is 360 minutes [](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes))
Then we tried to provide a timeout to the workflows, but it wasn't supported.
So I opened this PR to provide the inputs to be provided, with the default fallback.

Closes #78

### CODE REVIEW

File: .github/workflows/nx-cloud-main.yml [hunk 1]
```diff
@@ -79,6 +82,8 @@ jobs:
     # The name of the job which will invoke this one is expected to be "Nx Cloud - Main Job", and whatever we call this will be appended
     # to that one after a forward slash, so we keep this one intentionally short to produce "Nx Cloud - Main Job / Run" in the Github UI
     name: Run
+    # GitHub's default timeout is 360 minutes
+    timeout-minutes: ${{ inputs.timeout-minutes || 360 }}
```
Comment (meeroslav) [inline .github/workflows/nx-cloud-main.yml]: I would just call this property: "timeout"

File: .github/workflows/nx-cloud-agents.yml [hunk 2]
```diff
@@ -40,6 +40,12 @@ on:
       working-directory:
         required: false
         type: string
+      init-timeout-minutes:
```
Comment (meeroslav) [inline .github/workflows/nx-cloud-agents.yml]: We should use the same amount for the init as well as for the agent.

**Comment by meeroslav** [comment]
Thank you @johndalvik, this is a reasonable feature request.

Can you please address the two notes above and make sure to also modify Readme to include the new properties?

**Comment by johndalvik** [comment]
> Thank you @johndalvik, this is a reasonable feature request.
> 
> Can you please address the two notes above and make sure to also modify Readme to include the new properties?

@meeroslav Sure! I hope I understood your comments well, also updated the `README.md`.

**Review by meeroslav — APPROVED** [review]

**Comment by virtuoushub** [comment]
@johndalvik Just wanted to say thank you for adding this! ( and thanks @meeroslav for reviewing, approving and merging )

> We had a problem when running the workflows, one of the agents failed but somehow the main job stuck, ....
> Then we tried to provide a timeout to the workflows, but it wasn't supported. ...

_I also ran into this, tried to add `timeout-minutes` to the workflow's main/agents jobs, was unsuccessful, and came over to the docs to figure out if/how to configure._

_Was super easy for me to do, due to this PR._

**Pull request merged**